addons/l10n_vn_viettel/models/account_move.py

```diff
@@ -0,0 +1,398 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+# Written with the help from Komit
+
+import json
+import uuid
+from datetime import date, datetime
+from dateutil.relativedelta import relativedelta
+from num2words import num2words
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+from odoo.tools import float_compare
+
+import logging
+_logger = logging.getLogger(__name__)
+import urllib3
+try:
+    import certifi
+except ImportError:
+    certifi = None
+    _logger.warning(_('Could not import certifi. Please see https://pypi.org/project/certifi/. SSL verification is disabled for communications with Viettel.'))
+from odoo import tools
+
+
+inv_fields_mapping = {
+        "invoiceIssuedDate": "create_date",
+        "invoiceNote": "summary",
+        "buyerName": "customer_name",
+        "buyerLegalName": "customer_legal_name",
+        "buyerTaxCode": "customer_tax_code",
+        "buyerAddressLine": "customer_address",
+        "buyerEmail": "customer_email",
+        "transactionUuid": "transaction_uuid",
+        # seller info
+        "sellerLegalName": "company_name",
+        "sellerTaxCode": "company_tax_code",
+        "sellerAddressLine": "company_address",
+        "sellerPhoneNumber": "company_number",
+        "sellerWebsite": "company_website",
+        "sellerBankAccount": "company_bank_account",
+        "sellerBankName": "company_bank_name",
+
+        "paymentMethodName": "payment_method_name",
+        # summary
+        "sumOfTotalLineAmountWithoutTax": "total_line_amount_wo_tax",
+        "totalAmountWithoutTax": "amount_untaxed",
+        "totalTaxAmount": "amount_tax",
+        "totalAmountWithTax": "amount_total",
+        "totalAmountWithTaxInWords": "amount_total_text",
+        "taxPercentage": "total_tax_percent",
+        "taxBreakdowns": "tax_list",
+}
+
+inv_line_fields_mapping = {
+    "lineNumber": "sequence",
+    "itemCode": "product_ref",
+    "itemName": "product_name",
+    "unitName": "uom_name",
+    "unitPrice": "price_unit",
+    "quantity": "quantity",
+    "itemTotalAmountWithoutTax": "price_subtotal",
+    "taxAmount": "amount_tax",
+    "taxableAmount": "taxable_amount",
+    "taxPercentage": "tax_percent",
+}
+
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    payment_journal_id = fields.Many2one('account.journal', domain="[('type', 'in', ('bank', 'cash')), ('bank_account_id', '!=', False)]", string="Payment Journal", states={'open': [('readonly', True)]})
+    l10n_vn_transaction_id = fields.Char(copy=False)
```

Comment:
Because Odoo's naming convention is ****** and this is the transaction's id, whereas `record.account_id``is not the account's id but the account's instance. It is only the account's id in the database, and I think we are smart enough to know that the column account in SQL doesn't represent something else that the id.